llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2518,6 +2567,10 @@ class VPDerivedIVRecipe : public VPSingleDefRecipe {
   /// for floating point inductions.
   const FPMathOperator *FPBinOp;
 
+  VPCanonicalIVPHIRecipe *getCanonicalIV() const {
```

Comment:
After this patch, it/s not guaranteed that the second operand is the canonical IV. Better to completely remove the helper. Also, the documentation for the recipe needs updating above.